Add mostly failing tests for pickling various objects in plasmapy.particles

namurphy opened the pull request:
This PR adds new tests in `plasmapy.particles` regarding the pickling of particles.  Both `CustomParticle` and `DimensionlessParticle` instances can be pickled, but `Particle` instances (and objects that contain `Particle` instances) cannot.  Hence, most of these tests are marked as failing.  These tests should help later on when we address #1011 and make particles piclkealble.  (Yeah, I gave up on trying to spell that.)

StanczakDominik reviewed (commented):
Good test! I've simplified the tmp file a little, outsourcing that job to pytest (docs.pytest.org/en/stable/tmpdir.html), and then I added a test that verifies that these particles can then be loaded back from their pickles. Looks like that breaks for Custom and Dimensionless particles, and we should probably investigate that first...

namurphy commented:
> Good test! I've simplified the tmp file a little, outsourcing that job to pytest (docs.pytest.org/en/stable/tmpdir.html)

Thanks!  Good idea!

> and then I added a test that verifies that these particles can then be loaded back from their pickles. Looks like that breaks for Custom and Dimensionless particles, and we should probably investigate that first...

I changed `open(..., "r")` to `open(..., "rb")` since pickle files are binary.

StanczakDominik commented:
Oh bloody hell, i completely forgot about that, duh...
